metapool/sample_sheet.py

```diff
@@ -645,166 +985,49 @@ def make_sample_sheet(metadata, table, sequencer, lanes, strict=True):
     SampleSheetError
         If one of the required columns is missing.
     """
-    messages = _validate_sample_sheet_metadata(metadata)
+    required_attributes = ['SheetType', 'SheetVersion', 'Assay']
+
+    for attribute in required_attributes:
+        if attribute not in metadata:
+            raise ValueError("'%s' is not defined in metadata" % attribute)
+
+    sheet_type = metadata['SheetType']
+    sheet_version = metadata['SheetVersion']
+    assay_type = metadata['Assay']
+
+    if sheet_type == _STANDARD_SHEET_TYPE:
+        if assay_type == _AMPLICON:
+            sheet = AmpliconSampleSheet()
+        elif assay_type == _METAGENOMIC:
+            if sheet_version == '90':
+                sheet = MetagenomicSampleSheetv90()
+            elif sheet_version in ['95', '99', '100']:
+                # 95, 99, and v100 are functionally the same type.
+                sheet = MetagenomicSampleSheetv100()
+            else:
+                raise ValueError(f"'{assay_type}' is an unrecognized Sheet"
+                                 f"Version for '{sheet_type}'")
+        elif assay_type == _METATRANSCRIPTOMIC:
+            sheet = MetatranscriptomicSampleSheet()
+        else:
+            raise ValueError("'%s' is an unrecognized Assay type" % assay_type)
+    elif sheet_type == _ABSQUANT_SHEET_TYPE:
+        sheet = AbsQuantSampleSheetv10()
+    else:
+        raise ValueError("'%s' is an unrecognized SheetType" % sheet_type)
+
+    messages = sheet._validate_sample_sheet_metadata(metadata)
 
     if len(messages) == 0:
-        sheet = KLSampleSheet()
-        sheet = _add_metadata_to_sheet(metadata, sheet, sequencer)
-        sheet = _add_data_to_sheet(table, sheet, sequencer, lanes,
-                                   metadata['Assay'], strict)
+        sheet._add_metadata_to_sheet(metadata, sequencer)
+        sheet._add_data_to_sheet(table, sequencer, lanes, metadata['Assay'],
+                                 strict)
         return sheet
     else:
         for message in messages:
```

Comment:
if the above calls `validate_and_scrub_sample_sheet` then would the messages be echo'd twice?